### PR TITLE
Cloudwatch: Add Documentation on Temporary Credentials

### PR DESCRIPTION
**Pull request by sarahzinger**
**What is this feature?**

Documenting our new Temporary Credentials feature currently in Private Preview in Grafana Cloud

**Why do we need this feature?**

Currently all Grafana Cloud customers who want to access the CloudWatch Data source are required to create and rotate Secret and Access keys for all of their CloudWatch Datasource instances in Grafana, which can sometimes be a bit of a chore/pain point.

This feature instead allows customers to create IAM Roles that permit the Grafana Account to assume with STS. Grafana will then get temporary credentials to access customer data. 

**Who is this feature for?**

Grafana Cloud Customers

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:164]:
```suggestion
1. Put in a request to Customer Support to enable`awsDatasourcesTempCredentials`.
2. Once the feature is enabled, create a new CloudWatch data source (or update an existing one) and select `Grafana Assume Role` as an authentication provider.
3. Create a new IAM role in the AWS console, and select **Another AWS account** as the Trusted entity.
4. Enter the account ID of the Grafana account that has permission to assume this role 008923505280 and check the **Require external ID** box.
5. Enter the external ID specified in the instructions box on the **Settings** tab of CloudWatch data source configuration in Grafana. This external ID will be unique to your Grafana instance.
6. Attach any required permissions you would like Grafana to be able to access on your behalf (for example, CloudWatch Logs and CloudWatch Metrics policies).
7. Give the role a name and description, and click **Create role**.
8. Copy the ARN of the role you just created and paste it into the **Assume Role ARN** field on the **Settings** tab of CloudWatch data source configuration in Grafana.
```

Comment (imatwawana) [reply docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:164]: Some questions:

1. In step 3 is Trusted entity a field name?
2. What's the number in step 4?
3. On step 5, is this a titled part of the UI or just a place where there are instructions?

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [reply docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:151]: We should replace this with a private preview shortcode for the correct text

**Review by imatwawana — COMMENTED** [review]
Thanks for the contribution! I've reviewed the What's new and made a number of edits to the AWS docs, though for the data source docs it's mostly for styling. Once we've sorted out changes from this round of review, I'll tag in our data sources writer to do another pass since I'd like to get her opinion on the data source documentation. Please tag me for review again when you're ready for me to take another look. Also, make sure to view from the [**Files** tab of the PR](https://github.com/grafana/grafana/pull/75178/files) to ensure you catch all the suggestions.

File: docs/sources/whatsnew/whats-new-next/index.md
Comment (sarahzinger) [reply docs/sources/whatsnew/whats-new-next/index.md:92]: eeek nope that's a lingering merge conflict I missed thank you!

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:54]:
```suggestion
- `Grafana Assume Role` - With this auth provider option, Grafana Cloud users create an AWS IAM role that has a trust relationship with Grafana's AWS account. Grafana then uses STS to generate temporary credentials on its behalf. Users with this option enabled no longer need to generate secrets and access keys for users.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:67]:
```suggestion
When this field is filled in, Grafana uses the provided credentials to perform an [sts:AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) call. In this scenario, the primary authentication method does not need permission to access CloudWatch directly; it just needs the ability to assume a role, while the role it assumes should have permission to access CloudWatch.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:69]:
```suggestion
For example, you may have one set of long term credentials for all of your AWS data sources. However, you want to limit the access each data source has to AWS (maybe one accesses staging data and another production data, for example). You could create separate credentials for each data source, each maintaining its own set of permissions to various resources. However, depending on how many data sources you have and how you've set them up, that might mean rotating/managing many different secret and access keys. 
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:89]:
```suggestion
When `Grafana Assume Role` is the selected authentication provider, Grafana is responsible for supplying and calling the external ID. It's displayed on the data source configuration page and is unique to your account. It's very important when creating an IAM role for `Grafana Assume Role` that you set a conditional that Grafana's AWS account can only assume your IAM role when a particular external ID is specified:
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:164]: A couple questions:
In step 3, is Trusted entity a field label?
In step 4, what is the number after the word "role"?

I've made edits based on the "Trusted entity" being a field label and the number being a mistake.

```suggestion
3. Create a new IAM role in the AWS console, and select **Another AWS account** as the **Trusted entity**.
4. Enter the account ID of the Grafana account that has permission to assume this role and check the **Require external ID** box.
```

**Review by imatwawana — COMMENTED** [review]
This is looking solid! I've left a couple questions and now tagging @lwandz13 for review. @lwandz13, I wasn't 100% sure about the formatting of the Grafana Assume Role in the CloudWatch docs, so I'm leaving that up to you. I've done a general copyedit already, but you may find stuff I've missed. I'll take one last look when you're finished reviewing.

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (sarahzinger) [reply docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md]: Yeah, I think I'd describe this as a label:
Screenshot 2023-09-27 at 2 55 58 PM

and yeah sorry that last sentence got a bit lost from me I think. The number refers to our accountid. It will be the same for everyone which is why I put it in our documentation originally but I have removed it just to be a bit more future-proof in case we update it and forget to update the documentation. hopefully this is more clear!

File: docs/sources/whatsnew/whats-new-next/index.md
Comment (sarahzinger) [reply docs/sources/whatsnew/whats-new-next/index.md]: Realizing I didn't respond to this comment, sorry!

Spelled out STS, although like many aws acronyms it's hard to say if people are more familiar with the acronym or what it stands for haha. I think people who concern themselves with aws and auth will be pretty familar it's not that unique to grafana or anything. 

I'm not sure I totally follow the second question. 
Maybe this helps?

There's always been an assume role option available that looks like this: 
Screenshot 2023-09-27 at 3 25 03 PM

Users have been able to have their primary aws credentials assume a role for a long time. I think this was primarily used in OSS and On-Prem environments and mostly using the aws-sdk which would allow them to attach credentials directly to whatever machine was running Grafana (like EC2). That said it has also possible to use Assume role in cloud, they would just also need to provide secret and access keys first:
Screenshot 2023-09-27 at 3 27 20 PM
which kind of takes away from some of the main benefits of using an iam role in the first place.

So now we have a new feature that we're calling Grafana Assume Role, that lets Grafana Cloud users specify an ARN without giving us secret keys (kind of like the AWS SDK Default option above)
Screenshot 2023-09-27 at 3 24 02 PM

The ARN field doesn't change but the authentication provider does. 

I know this is super confusing because it's all called arn assume role haha, so I'm trying to make it more distinct/understandable

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:161]:
```suggestion
3. In the AWS Console, create a new IAM role, and under **Trusted entity type**, select **Another AWS account** as the trusted Entity.
```

File: docs/sources/whatsnew/whats-new-next/index.md
Comment (imatwawana) [inline docs/sources/whatsnew/whats-new-next/index.md:88]:
```suggestion
The Grafana Assume Role authentication provider lets Grafana Cloud users of the CloudWatch data source authenticate with AWS without having to create and maintain long term AWS Users. Using the new assume role authentication method, you no longer have to rotate access and secret keys in your CloudWatch data source. Instead, Grafana Cloud users can create an identity access and management (IAM) role that has a trust relationship with Grafana's AWS account; Grafana's AWS account will then use AWS Secure Token Service (STS) to create temporary credentials to access the user's AWS data.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:60]:
```suggestion
Assume a role is required for the Grafana Assume Role.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:71]:
```suggestion
The Grafana Assume Role also helps facilitate this. Using this role, Grafana's AWS account acts as the primary credential, having only the permission to assume roles in other accounts. You can then create IAM roles for Grafana's account to assume. For more information, refer to [Use Grafana assume role]({{< relref "#use-grafana-assume-role" >}}).
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:73]:
```suggestion
If the **Assume Role ARN** field is left empty, Grafana uses the provided credentials from the selected authentication method directly, and permissions to AWS data must be attached directly to those credentials. The **Assume Role ARN** field is optional for all authentication methods except for Grafana Assume Role.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:80]:
```suggestion
You cannot use an external ID for the Grafana Assume Role authentication provider.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:87]:
```suggestion
When Grafana Assume Role is the selected authentication provider, Grafana is responsible for supplying and calling the external ID. It's displayed on the data source configuration page and is unique to your account. It's very important when creating an IAM role for `Grafana Assume Role` that you set a conditional that Grafana's AWS account can only assume your IAM role when a particular external ID is specified:
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:149]:
```suggestion
## Use Grafana Assume Role
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:155]:
```suggestion
The Grafana Assume Role authentication provider lets you authenticate with AWS without having to create and maintain long term AWS users or rotate their access and secret keys. Instead, you can create an IAM role that has permissions to access CloudWatch and a trust relationship with Grafana's AWS account. Grafana's AWS account then makes an STS request to AWS to create temporary credentials to access your AWS data. It makes this STS request by passing along an `externalID` that's unique per Cloud account, to ensure that Grafana Cloud users can only access their own AWS data. For more information, refer to the [AWS documentation on external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html).
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:160]:
```suggestion
2. Once the feature is enabled, create a new CloudWatch data source (or update an existing one) and select **Grafana Assume Role** as an authentication provider.
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:157]:
```suggestion
To use the Grafana Assume Role:
```

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [reply docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md]: Thanks for circling back! I've made a bit of an update to step and left step 4 as is!

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:51]: In this case I'd leave it in code font just to be consistent with the other list items; but there's an argument to be made that none of these list items should be in code font.
```suggestion
- `Grafana assume role` - With this auth provider option, Grafana Cloud users create an AWS IAM role that has a trust relationship with Grafana's AWS account. Grafana then uses [STS](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) to generate temporary credentials on its behalf. Users with this option enabled no longer need to generate secret and access keys for users.
```

File: docs/sources/whatsnew/whats-new-next/index.md
Comment (imatwawana) [reply docs/sources/whatsnew/whats-new-next/index.md]: Thanks for the clarification! To my mind, based on your explanation and our style guidelines, the term should just be  plain text except when we're referencing it specifically in the UI (then it's bold and capitalized as it is in the UI; which I've done in one instance in the docs). As for the capitalization, I'm iffy. @lwandz13, as this is your area, I'll leave the final decision on that to you, but I've made suggestions for it to remove the code formatting on the term in the PR and you can take or leave those suggestions as you see fit.

File: docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md
Comment (imatwawana) [inline docs/sources/datasources/aws-cloudwatch/aws-authentication/index.md:51]:
```suggestion
- `Grafana Assume Role` - With this auth provider option, Grafana Cloud users create an AWS IAM role that has a trust relationship with Grafana's AWS account. Grafana then uses [STS](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) to generate temporary credentials on its behalf. Users with this option enabled no longer need to generate secret and access keys for users.
```

**Review by imatwawana — APPROVED** [review]
This is good to go for me! Just waiting on Larissa's review.

**Review by lwandz13 — APPROVED** [review]

**Pull request merged**
